includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -151,17 +151,30 @@ public function need_update_idempotency_key( $source_object, $error ) {
 		);
 	}
 
+	/**
+	 * Checks if keys are set.
+	 *
+	 * @since 4.0.6
+	 * @return bool
+	 */
+	public function are_keys_set() {
+		if ( $this->testmode ) {
+			return preg_match( '/^pk_test_/', $this->publishable_key )
+				&& ( preg_match( '/^sk_test_/', $this->secret_key ) || preg_match( '/^rk_test_/', $this->secret_key ) );
```

Comment:
Could this pattern (and the other equivalents) just be `/^[sr]k_test_/` or is that less readable?

Reply:
It's readable enough. I just copy&pasted and didn't think about it :)

In fact, that change would justify using regexps instead of a simpler `substr` call.

Changed it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1166/commits/151366ff430395af440ef88e938d098fa85233c5